### PR TITLE
mv: Mv hardlinks windows

### PR DESCRIPTION
**Pull request by dmatos2012**
Hi, 

This PR is related to #5451, but not necessarily closes it. On windows, making a hardlink would not be recognized, but now it is:

```
# on windows
create file named hard_link_a
mkhardlink /H hard_link_b hard_link_a
```
Currently:
```
> mv hard_link_a hard_link_b 
# removes hard_link_a 
```
Now:
```
> mv hard_link_a hard_link_b
mv: 'hard_link_a' and 'hard_link_b' are the same file 
```

Now with the issue related on case-insensitive OSs. Currently
```
touch a
mv a A
```
would succeed only on linux, but fail on both windows, macOs due to being case-insensitive Os. I have been trying to make it work across all OS, but changing the order of some of the error handling blows up some of the tests. Is that even something we want to support, or is the current behavior the intended one?

 I have seen lots of issues with `mv` lately, including some I caused with my PR (sorry 😛  ) , so would like to improve at least the behavior of certain operations.  

Basically I started trying to fix case-insensitive, ended up fixing moving hardlinks :P

### CODE REVIEW

**Comment by fdncred** [comment]
What can we do to get this going? I believe it may be the only thing holding up integration of uu_mv into nushell. Any ideas @tertsdiepraam? Sorry to tag you. It's been a few weeks and it would be great to move forward with this.

**Comment by tertsdiepraam** [comment]
I'll take a look :)

File: src/uu/mv/src/mv.rs [hunk 1]
```diff
@@ -325,8 +326,9 @@ fn handle_two_paths(source: &Path, target: &Path, opts: &Options) -> UResult<()>
         });
     }
 
+    let is_source_same_target = is_same_file(source, target).unwrap_or(false);
```
Comment (tertsdiepraam) [inline src/uu/mv/src/mv.rs:329]: I wonder if `is_same_file` is already doing the same checks that we do in the Unix version of `are_hardlinks_to_same_file`. Can we simply use `is_same_file` for both platforms?

File: tests/by-util/test_mv.rs [hunk 2]
```diff
@@ -1443,6 +1444,29 @@ fn test_mv_dir_into_file_where_both_are_files() {
...
+fn test_mv_same_case_insensitive_file() {
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+    let file1 = "a.txt";
+    at.touch(file1);
+    #[cfg(target_os = "linux")]
+    scene.ucmd().arg(file1).arg("-f").arg("A.txt").succeeds();
+    #[cfg(not(target_os = "linux"))]
+    scene.ucmd().arg(file1).arg("-f").arg("A.txt").fails();
+}
```
Comment (tertsdiepraam) [inline tests/by-util/test_mv.rs:1457]: Very very very minor nit, but could you add a newline here? 😄

File: tests/by-util/test_mv.rs [hunk 3]
```diff
@@ -1443,6 +1444,29 @@ fn test_mv_dir_into_file_where_both_are_files() {
         .stderr_contains("mv: cannot stat 'a/': Not a directory");
 }
 
+#[test]
+fn test_mv_same_case_insensitive_file() {
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+    let file1 = "a.txt";
+    at.touch(file1);
+    #[cfg(target_os = "linux")]
```
Comment (tertsdiepraam) [inline tests/by-util/test_mv.rs:1453]: Are the BSDs case-insensitive as well? I think this might actually also be more complicated because this might also depend on the file system. I'm not entirely sure though.

**Review by tertsdiepraam — COMMENTED** [review]
Thanks for the ping here!

File: src/uu/mv/src/mv.rs (hunk 1 above)
Comment (dmatos2012) [reply src/uu/mv/src/mv.rs:329]: Yes, I think `is_same_file` is doing the same checks, so it could be used in all platforms, but was not sure to change all of that code before confirmation.

File: tests/by-util/test_mv.rs (hunk 3 above)
Comment (dmatos2012) [reply tests/by-util/test_mv.rs:1453]: Yeah this is also where im a bit out of my league, because handling cross-platforms is so difficult, specially given that I cant test what is working or not. I am really wondering if we should even have different behaviors on different platforms, which has been my biggest issue :(. Thats why I have left it to @fdncred and other more smart people to figure out way forward

Comment (tertsdiepraam) [reply tests/by-util/test_mv.rs:1453]: I'd like to see at least the FreeBSD CI pass (it seems to be cancelled on the last run). Then we'll just see if anyone complains 😄

Comment (fdncred) [reply tests/by-util/test_mv.rs:1453]: I'd vote for consistency across all platforms as much as possible.

**Comment by sylvestre** [comment]
please reopen when ready
